site/cypress/e2e/Cypress-Gradeable/office_hours_queue.spec.js

```diff
@@ -0,0 +1,213 @@
+const queueName = 'Cypress Office Hour Queue 1';
+const queueName_random = 'Cypress Office Hour Queue Random';
+const queueName1 = 'Cypress Office Hour Queue 2';
+const queueCode = 'cypress_test';
+const queueCode1 = 'cypress_test_fail';
+const newQueueCode = 'cypress_update';
+const enableQueue = () => {
+    cy.visit(['sample']);
+    cy.get('#nav-sidebar-course-settings').click();
+    cy.get('#queue-enabled').check();
+    cy.get('#queue-enabled').should('be.checked');
+};
+const deleteQueue = () => {
+    cy.visit(['sample']);
+    cy.get('#nav-sidebar-queue').click();
+    cy.get('#toggle_filter_settings').click();
+    cy.get('[data-testid="popup-window"]').should('exist');
+    cy.get('.delete_queue_btn').last().click();
+};
+const disableQueue = () => {
+    cy.logout();
+    cy.login();
+    cy.visit(['sample']);
+    cy.get('#nav-sidebar-course-settings').click();
+    cy.get('#queue-enabled').should('be.checked');
+    cy.get('#queue-enabled').check();
+};
+const openNewQueue = (queueName, queueCode='') => {
+    cy.get('#nav-sidebar-queue').click();
+    cy.get('#toggle_new_queue').should('exist').click();
+    cy.get('[data-testid="popup-window"]').should('exist');
+    cy.get('#new_queue_code').type(queueName);
+    if (queueCode.length > 0) {
+        cy.get('#new-queue-token').type(queueCode);
+    }
+    else {
+        cy.get('#new_queue_rand_token').click();
+    }
+    cy.get('#open_new_queue_btn').click();
+};
+
+const changeQueueCode = (queueName, queueCode='') => {
+    cy.get('#toggle_filter_settings').should('exist').click();
+    cy.get('[data-testid="popup-window"]').should('exist');
+    cy.get('#old_queue_code').select(queueName);
+    if (queueCode.length > 0) {
+        cy.get('#old_queue_token').type(queueCode);
+    }
+    else {
+        cy.get('#old_queue_rand_token').click(); //random
+    }
+    cy.get('#change_code_btn').click(); // update it
+};
+const expectedAlerts =  (success=0, error=0, success_text='', error_text='') => {
+    if (success > 0) {
+        cy.get('.alert-success').contains(success_text);
+    }
+    if (error > 0) {
+        cy.get('.alert-error').contains(error_text);
+    }
+};
```

Comment:
I think that this function is excessive, you should be able to just use the single line statements instead of having this function.